src/entities/schedulers/mempool-checker.js

```diff
@@ -0,0 +1,69 @@
+// @flow
+
+import _ from 'lodash'
+import type { Logger } from 'bunyan'
+import { helpers } from 'inversify-vanillajs-helpers'
+
+import type { RawDataProvider, Database } from '../../interfaces'
+import { TX_STATUS } from '../../blockchain/common'
+import SERVICE_IDENTIFIER from '../../constants/identifiers'
+import type { ShelleyTxType } from '../../blockchain/shelley/tx'
+import { sleep } from '../../utils'
+
+import BaseScheduler from './base-scheduler'
+
+class MempoolChecker extends BaseScheduler {
+  dataProvider: RawDataProvider
+
+  checkMempoolMillis: number
+
+  db: Database<ShelleyTxType>
+
+  constructor(
+    checkMempoolSeconds: number,
+    logger: Logger,
+    dataProvider: RawDataProvider,
+    db: Database<ShelleyTxType>,
+  ) {
+    super(logger)
+    this.checkMempoolMillis = checkMempoolSeconds * 1000
+    this.dataProvider = dataProvider
+    this.db = db
+
+    logger.debug('[MempoolChecker] Checking mempool every', checkMempoolSeconds, 'seconds')
+  }
+
+  async updateFailedTxs(txHashes: Array<string>): Promise<void> {
+    if (_.isEmpty(txHashes)) {
+      return
+    }
+    await this.db.updateTxsStatus(txHashes, TX_STATUS.TX_FAILED_STATUS)
+    for (const txHash of txHashes) {
+      await this.db.addNewTxToTransientSnapshots(txHash, TX_STATUS.TX_FAILED_STATUS)
+    }
+  }
+
+  async startAsync() {
+    this.logger.info(`[${this.name}] starting checking mempool for failed transactions.`)
+    for (;;) {
+      const failedTxs = (await this.dataProvider.getMessagePoolLogs())
+        .filter(msg => msg.status.Rejected !== undefined)
+        .map(msg => msg.fragment_id)
+      await this.updateFailedTxs(failedTxs)
```

Comment:
In the `.map(msg => msg.fragment_id)` we are losing the Jormun config on the rejection reason. Would be great if we could retain it here and at least log some messages like:
```
TX_REJECTED: { id=${fragment_id}, reason=${status.Rejected} }
```